x-pack/platform/plugins/shared/streams/server/routes/streams/significant_events/route.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { badRequest } from '@hapi/boom';
+import { SignificantEventsGetResponse } from '@kbn/streams-schema';
+import { z } from '@kbn/zod';
+import { createServerRoute } from '../../create_server_route';
+import { readSignificantEvents } from './read_significant_events';
+
+const stringToDate = z.string().transform((arg) => new Date(arg));
+
+export const readSignificantEventsRoute = createServerRoute({
+  endpoint: 'GET /api/streams/{name}/significant_events 2023-10-31',
```

Comment:
@flash1293 any opinions on making these public vs internal?

Reply:
Good question, I'd say public since there is a good use case for people to integrate with this via API (while there isn't really for things like the sample APIs). To me it's basically CRUD